src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/VertexAIEmbeddingClient.java

```diff
@@ -29,24 +31,86 @@ public interface VertexAIEmbeddingService {
     @POST
     @Path("/{modelId}:predict")
     @ClientHeaderParam(name = "Content-Type", value = "application/json")
-    List<float[]> embed(
+    EmbeddingResponse embed(
         @HeaderParam("Authorization") String accessToken,
         @PathParam("modelId") String modelId,
         EmbeddingRequest request);
   }
 
-  private record EmbeddingRequest(List<Content> instances, Options options) {
+  private record EmbeddingRequest(List<Content> instances) {
     public record Content(String content) {}
+  }
+
+  private static class EmbeddingResponse {
+    public EmbeddingResponse() {}
+
+    private List<Prediction> predictions;
+
+    @JsonIgnore private Object metadata;
+
+    public List<Prediction> getPredictions() {
+      return predictions;
+    }
+
+    public void setPredictions(List<Prediction> predictions) {
+      this.predictions = predictions;
+    }
+
+    public Object getMetadata() {
+      return metadata;
+    }
+
+    public void setMetadata(Object metadata) {
+      this.metadata = metadata;
+    }
+
+    private static class Prediction {
```

Comment:
Probably better to keep package protected (no modifier) if and when this needs to be serialized; Jackson can access private classes too, but has to force access and there's (I think) minor performance penalty (not enough to be measurable probably tho).